app/experimenter/docs/adrs/0005-dto-app-identifiers.md

```diff
@@ -0,0 +1,240 @@
+
+**<u>In the DTO,</u>** we previously mapped application constants to
+**application** but as of
+[<u>\#4239</u>](https://github.com/mozilla/experimenter/pull/4239) we
+are sending channel constants for mobile in the application field.
+
+### X-Man
+
+X-Man does not use the **application** or **channel** property in the
+DTO. X-Man assumes all experiments in the desktop remote settings
+collection might be relevant, and then evaluates the JEXL **targeting**
+property to determine eligibility, which includes the
+normalized\_channel.
+
+### Rust SDK
+
+The SDK uses **application** property in the DTO for targeting and the
+[<u>SDK
+expects</u>](https://github.com/mozilla/experimenter/issues/4235) the
+application field to exactly match the app store bundle identifier of
+the implementing app.
+
+### Jetstream
+
+Jetstream currently ignores both **application** and **channel** fields
+and assumes that all experiments target Desktop. Ideally this will
+change in the immediate future\! Jetstream needs to know:
+
+  - > which application family (i.e. “Fenix” or “Lockwise”) the
+    > experiment is targeting, in order to choose the right default set
+    > of metrics
+
+  - > which BigQuery dataset to find the app’s data in. For Glean
+    > telemetry, this is a string transformation of the bundle
+    > identifier. Each release channel of the Firefox for Android and
+    > iOS apps has a unique bundle identifier.
+
+The data pipeline does not currently maintain a mapping of application
+family, release channel, and bundle ID anywhere. One partial solution is
+a mapping from (legacy telemetry app\_name, normalized\_os) to a product
+name and canonical name in the [<u>product\_info
+UDF</u>](https://github.com/mozilla/bigquery-etl/tree/master/sql/mozfun/norm/product_info),
+which has a notion of product. The [<u>probe scraper repository
+list</u>](https://probeinfo.telemetry.mozilla.org/glean/repositories)
+enumerates bundle IDs and describes their associated release channel but
+doesn’t have any unifying concept of product family to link the
+different release channels together, although this is [<u>likely to
+change
+soon</u>](https://docs.google.com/document/d/1_zx0cxBnnFOhct1cCb_QnXntV-HN9sjqfGfiBwEBL-8/edit?ts=5fd8dab1#heading=h.151eux68loqq).
+
+# Decision drivers
+
+  - > Avoid standing up new information services: ideally each consumer
+    > will be able to derive all of the context they need from the
+    > payload
+
+  - > Align values and labels with other systems where possible
+
+  - > Handle quirks of existing and known-future (i.e. desktop Glean)
+    > consumers
+
+  - > Allow Experimenter to use the same mechanisms for channel
+    > selection across platforms someday: it will simplify the
+    > recipe-writing logic if we can rely on the metadata fields in the
+    > DTO to describe channel targeting on all platforms
+
+# Decision outcome (proposed)
+
+Option 1b is selected because it aligns well with other systems, gives
```

Comment:
Personally I'd lead with the summary of the decision here for readers who want the TLDR; or at else put a decision summary up at the very top i.e. 

```md
We decided on [Option 1b](link goes here), i.e.:
 - **app_name** will represent a notion of application family, like “Firefox,” “Fenix,” “Firefox iOS,” “Focus Android,” etc.
 - **app_id**  represent the app bundle ID (e.g. `org.mozilla.firefox`) or `firefox-desktop` on desktop), which can be used by both the SDK for targeting and Jetstream for deciding the right glean datasets.
 - **channel** is like “release,” “beta,” etc (even on mobile)

We selected this because....
```